### ISSUE TITLE
Wrong timestamps

### ISSUE THREAD

**Issue by MrChoclate** [cmusphinx/pocketsphinx#73]
Hello,

I used `pocketsphinx_continuous -infile ... -time yes` and the times are not always correct. I am not talking about more or less 10ms but more or less 1s. Sometimes 3s !
When this happened, there is a rollback in time for the next sentence.

For example, I got this log by processing http://ia800208.us.archive.org/8/items/red_house_mystery_librivox/redhouse_14_milne.mp3:
```
could you know i'd said del turning around to the stock tiny
<s> 228.760 228.830 0.999100
could 228.840 229.000 0.451901
you 229.010 229.080 0.088897
know 229.090 229.370 0.116595
i'd 229.380 229.610 0.073102
<sil> 229.620 229.820 0.998102
<sil> 229.830 230.070 0.999800
said 230.080 230.380 0.455258
del 230.390 230.690 0.013421
turning 230.700 231.060 0.205690
around(2) 231.070 231.330 0.201013
to(2) 231.340 231.430 0.039856
the 231.440 231.520 0.802659
stock 231.530 232.040 0.377112
<sil> 232.050 232.280 0.862503
tiny 232.290 232.980 0.028943
</s> 232.990 233.220 1.000000
if he was not so quick and thoughts on t he was quick enough in action
<s> 232.760 232.850 0.999200
if 232.860 232.990 0.885313
he 233.000 233.080 0.598003
```
The next sentence `if he was not so quick and thoughts on t he was quick enough in action` starts before (232.760 < 233.220) the end of the first one `could you know i'd said del turning around to the stock tiny`. The main problem is that the first sentence times are wrong. For example at 228.840 we hear "said Bill turning around" which should start at 230.080 according to sphinx.
I hope I clearly explained my problem.

This not a unique bug since I encountered this  ~100 times processing ~20h of audiobooks.

**Comment by nshmyrev** [comment]
Yes, it is sort of known issue. Not easy to fix, requires the total rewrite of the frontend.

**Comment by KuribohG** [comment]
Has this issue been fixed?

**Comment by iCreptio** [comment]
Any idea if this has been fixed?

**Comment by nshmyrev** [comment]
No

**Comment by iCreptio** [comment]
Any idea what is causing this by any chance? is it the same as https://gitmemory.com/issue/cmusphinx/pocketsphinx/170/481093289?

**Comment by nshmyrev** [comment]
The current implementation of the frontend is flawed, I wrote above. There is no much sense to fix it though, try kaldi instead.

**Comment by iCreptio** [comment]
Thank you - I'll check out kaldi, appreciate your work nevertheless!

**Comment by jsalsman** [comment]
@nshmyrev that's twice in three months you've recommended Kaldi. Are they shipping with open source models now? To @ChankiWu in https://github.com/cmusphinx/pocketsphinx/issues/188 too.

https://deepspeech.readthedocs.io/en/v0.6.0/DeepSpeech.html is so much better than Kaldi.

@brijmohan should we convert to DeepSpeech for phonological features? Would that solve our tonal languages bug? Can we still do it on the client, perhaps for a lot less battery?

**Comment by nshmyrev** [comment]
@jsalsman I'm suggesting the most accurate solution up to date. Kaldi models are available here:
https://github.com/alphacep/kaldi-android-demo/releases or via docker at https://github.com/alphacep/kaldi-websocket-python or some other at http://kaldi-asr.org/models.html

> https://deepspeech.readthedocs.io/en/v0.6.0/DeepSpeech.html is so much better than Kaldi.

Why do you think it is better? Accuracy wise it is about 2 times worse and there are many other issues.

**Comment by jsalsman** [comment]
Nickolay, I am thrilled to learn that Kaldi has a free English model!
Spectacular news and I will reconsider my opinion. A little less than a
year ago, DeepSpeech was claiming 18 languages, including tonal Mandarin,
to traditional Chinese graphemes. Now I know why Microsoft was in such
haste to check in https://github.com/kaldi-asr/kaldi/issues/3675

**Comment by nshmyrev** [comment]
Yes, things change pretty fast. Languages will be available more or less soon, probably faster than we think. I do not think Xingyu work was a haste, sorry, new code appears here and there.

**Comment by jsalsman** [comment]
Nickolay, again, great news. You pointed out the lack of context of
posterior phone probabilities alone suffers from conflating incidental and
consequential errors at least once in 2014, if I remember correctly. You
may have phrased it like, "acoustic scores don't work for confidence
values, and you can't simply scale them to fix it." It's not really a huge
problem if they want to keep doing that. Maybe when Mozilla Common Voice
adds transcripts instead of just booleans in Q2- we can provide authentic
intelligibility.

**Comment by nshmyrev** [comment]
@jsalsman Thank you James. I also look forward to see it happens!

**Comment by dhdaines** [comment]
Hi, the VAD has been removed from the frontend, because alignment is basically the most important application of pocketsphinx at the moment, not being a very good system for LVCSR.  So this ought to be fixed in #255
